.github/workflows/static-checks.yaml

```diff
@@ -0,0 +1,134 @@
+on:
+  workflow_call:
+    inputs:
+      runs-on:
+        required: true
+        type: string
+      gochannel:
+        required: true
+        type: string
+
+jobs:
+  static-checks:
+    runs-on: ${{ inputs.runs-on }}
+    env:
+      GOPATH: ${{ github.workspace }}
+      # Set PATH to ignore the load of magic binaries from /usr/local/bin And
+      # to use the go snap automatically. Note that we install go from the
+      # snap in a step below. Without this we get the GitHub-controlled latest
+      # version of go.
+      PATH: /snap/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:${{ github.workspace }}/bin
+      GOROOT: ""
+      GITHUB_PULL_REQUEST_TITLE: ${{ github.event.pull_request.title }}
+      BASE_REF: ${{ github.base_ref }}
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        # needed for git commit history
+        fetch-depth: 0
+        # NOTE: checkout the code in a fixed location, even for forks, as this
+        # is relevant for go's import system.
+        path: ./src/github.com/snapcore/snapd
+
+    # Fetch base ref, needed for golangci-lint
+    - name: Fetching base ref ${{ github.base_ref }}
```

Comment:
After looking through the workflows, I believe the most sensible common point is the "downloading cached dependencies, extracting them, and installing them" stages. I created a local action for those three steps. I didn't see any other obvious groups of steps that are shared between jobs. 

I noticed that the unit tests job also fetches the base ref, but I believe that is entirely unnecessary since the unit tests don't run golangci-lint, unless I'm mistaken, so instead of grouping that in with the Debian dependency steps, I would just delete it during the clean-up task.